ql/src/java/org/apache/hadoop/hive/ql/udf/UDFLike.java

```diff
@@ -77,9 +80,9 @@ public static String likePatternToRegExp(String likePattern) {
       if (n == '_') {
         sb.append(".");
       } else if (n == '%') {
-        sb.append(".*?");
+        sb.append(".*");
```

Comment:
I am not sure about this change, should we keep the matching lazy?

Reply:
Reverted. We only use the pattern for string matching, is reasonable to use the non-greedy matching.

Reply:
JDO does not support non-greedy matching :(

Add a new parameter `greedyMatch` for this option.